### ISSUE TITLE
feature request: make kopia UI portable

### ISSUE THREAD

**Issue by badrelmers** [kopia/kopia#368]
Hi,
is it possible to make Kopia UI portable? so we can put it an usb with it's config files.
thank  you

**Comment by jkowalski** [comment]
Great idea.

**Comment by jkowalski** [comment]
@badrelmers I have few questions:

1. Do you want to store the entire repo on USB stick (app + data) or just app to be able to access repo in the cloud?

2. Is it ok if the entire KopiaUI app is unpacked to TEMP directory and runs from there? That's how Electron portable apps are packaged - you get one EXE file that transparently unpacks itself, runs and deletes files.

3. Do you care if cache files are written to regular TEMP directory? They are encrypted.

**Comment by badrelmers** [comment]
> 1.  Do you want to store the entire repo on USB stick (app + data) or just app to be able to access repo in the cloud?

I want to store the entire repo on USB stick (app + data) and any config file that KopiaUI will need. 
Generally portable apps are a zip file, and when uncompressed and run from this folder it should create a folder inside this uncompressed folder that contains all the data and config files that otherwise would be saved in the home directory %APPDATA%\kopia-ui

> 2. Is it ok if the entire KopiaUI app is unpacked to TEMP directory and runs from there? That's how Electron portable apps are packaged - you get one EXE file that transparently unpacks itself, runs and deletes files.

I do not think this is a portable app, I see it like an ephemeral app, because that exe while it is portable it does not contain my config/data files

> 3. Do you care if cache files are written to regular TEMP directory? They are encrypted.

it is fine to save the cache in TEMP if the cache is auto created and do not break anything when I change the location of the portable folder. 

this is for me because I m the only one who use my computer, and I will be using this portable version in the same computer, because I format my computer (windows) a lot so having portable apps save a lot of time (to install and config).
**But generally I see a lot of people complain about saving the cache outside of the portable folder, and they consider it a fail.** 

> They are encrypted.

they are encrypted even when I choose to no encrypt the repository?

_______________________________
generally I saw most of portable apps in windows work like this:
when KopiaUI run it should search for a folder called **config** near to KopiaUI exe, if found then run the config file from this folder, if not found then use %APPDATA%\kopia-ui

N.B.:
I m already using KopiaUI in portable mode, I unzipped the exe with 7z and it works without any problem, my only problem is that I do not have any way to change the config folder (%APPDATA%\kopia-ui) (%APPDATA%\kopia)

**Comment by badrelmers** [comment]
@jkowalski 

Hi,

I want to add a finding which relate to this issue.

if I add this command line switch: `--user-data-dir='full\path'` to Kopia UI all the files in `%APPDATA%\kopia-ui` are saved in `full\path` including the `config.json`.

this make Kopia UI portable already, it works perfectly in my tests.

but this switch is not mentioned in Electron doc about supported [command-line-switches](https://www.electronjs.org/docs/api/command-line-switches)
but it I found that the switch was added/edited in this commit  https://github.com/electron/electron/commit/da58ac7c20f92973801f37cbcab6f430f4e5d045

so I think it is supported but not mentioned in the Doc because the doc is about v8.2.0 and the commit is for v10.0.0-nightly.20200327  v7.0.0-beta.1.

maybe I m wrong and it is/will not officially supported, but it works.
